Detectors/ZDC/simulation/src/digi2raw.cxx

```diff
@@ -0,0 +1,162 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// \file digi2raw.cxx
+/// \author ruben.shahoyan@cern.ch
+
+#include <boost/program_options.hpp>
+#include <TSystem.h>
+#include <TFile.h>
+#include <TStopwatch.h>
+#include "Framework/Logger.h"
+#include <string>
+#include <iomanip>
+#include "CCDB/BasicCCDBManager.h"
+#include "CCDB/CCDBTimeStampUtils.h"
+#include "CommonUtils/StringUtils.h"
+#include "CommonUtils/ConfigurableParam.h"
+#include "DetectorsCommonDataFormats/NameConf.h"
+#include "DetectorsRaw/HBFUtils.h"
+#include "ZDCBase/Constants.h"
+#include "ZDCBase/ModuleConfig.h"
+#include "DataFormatsZDC/BCData.h"
+#include "DataFormatsZDC/ChannelData.h"
+#include "ZDCSimulation/Digits2Raw.h"
+#include "DataFormatsParameters/GRPObject.h"
+#include "SimulationDataFormat/DigitizationContext.h"
+
+/// MC->raw conversion for ZDC
+
+namespace bpo = boost::program_options;
+
+void digi2raw(const std::string& inpName, const std::string& outDir, int verbosity, bool filePerLink, uint32_t rdhV = 4, bool noEmptyHBF = false,
+              int superPageSizeInB = 1024 * 1024);
+
+int main(int argc, char** argv)
+{
+  bpo::variables_map vm;
+  bpo::options_description opt_general("Usage:\n  " + std::string(argv[0]) +
+                                       "Convert ZDC digits to CRU raw data\n");
+  bpo::options_description opt_hidden("");
+  bpo::options_description opt_all;
+  bpo::positional_options_description opt_pos;
+
+  try {
+    auto add_option = opt_general.add_options();
+    add_option("help,h", "Print this help message");
+    add_option("verbosity,v", bpo::value<int>()->default_value(0), "verbosity level");
+    //    add_option("input-file,i", bpo::value<std::string>()->default_value(o2::base::NameConf::getDigitsFileName(o2::detectors::DetID::ZDC)),"input ZDC digits file"); // why not used?
+    add_option("input-file,i", bpo::value<std::string>()->default_value("zdcdigits.root"), "input ZDC digits file");
+    add_option("file-per-link,l", bpo::value<bool>()->default_value(false)->implicit_value(true), "create output file per CRU (default: per layer)");
```

Comment:
the copy-pasted comment is misleading. We must have at least an option to write 1 file per link. As far as I can see from your code, your default is to write just 1 file. So, the help comment should be ``create output file per link (default: write single file for links)``